docs/docs/guides/events_subscriptions/custom_subscriptions.md

```diff
@@ -0,0 +1,230 @@
+---
+sidebar_position: 2
+sidebar_label: 'Custom Subscriptions'
+---
+
+# Custom Subscription
+
+You can extend the `Web3Subscription` class to create custom subscriptions. This way you can subscribe to custom events emitted by the provider.
+
+:::note
+This guid is most likely for advanced users who are connecting to a node that provide additional custom subscription. And for normal users, the standard subscriptions are supported out of the box as you can find in [Supported Subscriptions](/guides/events_subscriptions/supported_subscriptions).
+:::
+
+:::important
+If you are the developer who provide custom subscriptions to users. We encourage you to develop a web3.js Plugin after you go through the guide below. However, you can find how to develop a plugin at [web3.js Plugin Developer Guide](/guides/web3_plugin_guide/plugin_authors).
+
+And even if you are not the developer how provider this custom subscriptions, we encourage you to write a web3.js plugin for the custom subscription, and publish it to the npm registry. This way you can help the community. And they might contribute to your repository helping for things like: feature addition, maintenance, and bug detection.
```

Comment:
And even if you are not the developer **how provider this custom subscriptions,** we encourage you to write a web3.js plugin for the custom subscription, 

This is not clear

Reply:
Sorry, there were 2 typos. However, I now made many edits to the whole file. And I hope it is much better now :smile: